src/main/java/org/gradle/profiler/studio/invoker/StudioSample.java

```diff
@@ -0,0 +1,34 @@
+package org.gradle.profiler.studio.invoker;
+
+import org.gradle.profiler.result.BuildInvocationResult;
+import org.gradle.profiler.result.Sample;
+
+import java.time.Duration;
+
+public enum StudioSample implements Sample<BuildInvocationResult> {
+    GRADLE_EXECUTION_TIME("Gradle execution time") {
+        @Override
+        public Duration extractFrom(BuildInvocationResult result) {
+            StudioBuildActionResult studioResult = (StudioBuildActionResult) result.getActionResult();
```

Comment:
We don't want to introduce a `StudioBuildInvocationResult` to avoid this cast?

Reply:
That would be ideal, but it requires some changes, I will check what I can do.

Reply:
What I did. I created a StudioBuildInvocationResult, and I do a cast there. It's not totally perfect, but I think that right now that is the simplest and cleanest solution. 

Otherwise, we would have to inject (or override some method that will return) a custom `BuildUnderTestInvoker` into `GradleScenarioInvoker` and also create a new `InvokeAndMeasureAction` that would return  `StudioBuildInvocationResult`. Additionally to that, `BuildAction` would have to return generic result from `run` method and that would require passing generics everywhere and it will get a bit messy.

WDYT?

Reply:
Alright, one step at a time.